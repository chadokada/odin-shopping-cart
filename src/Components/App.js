import React from 'react';
import '../Styles/App.css';
import  { Link } from 'react-router-dom';


function App() {
  return (
    <div className="App">
      <div className='store-name'>Butter and Sugar</div>
      <p>Hand-crocheted goods made with love and care in the heart of Paradise</p>
      <Link to="/all">
        <button className='shop-now-btn'>Shop Now</button>
      </Link>
    </div>
  );
}

export default App;
