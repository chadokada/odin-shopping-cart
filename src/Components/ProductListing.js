const products = { //Put this in a separate file 
  hats : [
    {name: 'Ribbed Beanie', price: 30.00, source: 'ribbed_beanie.jpg', type:'hats'},
    {name: 'Bowler', price: 35.00,source: 'bowler.jpg', type:'hats'},
  ],
  bags : [
    {name: 'Tote Bag', price: 40.00,source: 'crochet_tote.jpg', type:'bags'},
    {name: 'Grocery Bag', price: 30.00,source: 'grocery_bag.png', type:'bags'},
  ]
}

export {products};