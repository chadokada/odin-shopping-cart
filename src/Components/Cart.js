import React from 'react';
import '../Styles/App.css';

const Cart = ({showCart, toggleCart}) => {

  let cartOpen = '';
  if (showCart === true){
    cartOpen = ' open';
  }



  return (
    <div className={`cart-drawer${cartOpen}`}>
      THIS IS CARTA
      <button onClick={toggleCart}>Close Cart</button>
    </div>
  )
}

export default Cart;