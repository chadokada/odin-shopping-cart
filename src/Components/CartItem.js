import React from 'react';
import '../Styles/App.css';

const CartItem = ({product, decrementItem, incrementItem, handleQuantityInput}) => {
  const productName = product.name;
  const productType = product.type;
  const imageSrc = product.source;
  const price = product.price.toFixed(2);
  const quantity = product.quantity;

  return (
    <div 
      className='cart-item'
      name={productName}
      type={productType}
    >
      <div className='cart-item-img-container'>
        <img
          className='cart-item-img'
          alt={product}
          src={require(`../Images/Products/${imageSrc}`)}
        />
      </div>
      <div className='cart-item-side'>
        <div className='cart-item-info'>
          <div className='cart-item-name'>{productName}</div>
          <div className='cart-item-price'>{`$${price}`}</div>
        </div>

        <div className='cart-item-quantity-container'>
          <button onClick={decrementItem}>-</button>
          <input 
            type='number' 
            className='cart-item-quantity-input' 
            value={quantity}
            onInput={handleQuantityInput}
          >
          </input>
          <button onClick={incrementItem}>+</button>
        </div>
      </div>
    </div>
  )
}

export default CartItem;