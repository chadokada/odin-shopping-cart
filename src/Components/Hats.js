import React from 'react';
import '../Styles/App.css';


const Hats = () => {
  return (
    <div>
      Hats
    </div>
  )
}

export default Hats;