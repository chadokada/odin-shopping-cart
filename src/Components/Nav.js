import React from 'react';
import '../Styles/App.css';
import  { Link } from 'react-router-dom';

const Nav = ({shopName}) => {
  return (
    <nav className='nav'>
      <h2>{shopName}</h2>
      <ul>
          <Link to="/products/all">
            <li>All Products</li>
          </Link>
          <Link to="/products/hats">
            <li>Hats</li>
          </Link>
          <Link to="/products/bags">
            <li>Bags and Totes</li>
          </Link>
      </ul>
    </nav>
  )
}

export default Nav;