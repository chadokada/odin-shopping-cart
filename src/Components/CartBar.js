import React, {useState} from 'react';
import '../Styles/App.css';
import shoppingCart from '../Images/Icons/shopping-cart.png';
import Cart from './Cart';

const CartBar = (
    {
      cart, 
      decrementItem, 
      incrementItem, 
      handleQuantityInput, 
      handleCheckOut
    }
  ) => {

  const getCartSize = () => {
    let cartSize = 0;
    for (const item in cart) {
      cartSize += cart[item].quantity;
    }
    return cartSize;
  }  
  
  const cartSize = getCartSize();

  let cartStatus = ''
  if (cartSize > 0) {
    cartStatus = ' full'
  };

  let [showCart, setShowCart] = useState(false);

  const toggleCart = () => {
    if (showCart === false) {
      setShowCart(true)
    } else {
      setShowCart(false)
    }
  }

  return (
  <div className="top-bar">
    <div className='store-name'>Butter and Sugar</div>

    <div className={`shopping-cart${cartStatus}`} onClick={toggleCart} data-testid='cart-button'>
      <div className='cart-size' data-testid='cart-size'>{cartSize}</div>
      <img alt="cart-icon" src={shoppingCart} className={`cart-icon${cartStatus}`}/>
    </div>
    <Cart 
      cart={cart} 
      showCart={showCart} 
      toggleCart={toggleCart}
      decrementItem={decrementItem}
      incrementItem={incrementItem}
      handleQuantityInput={handleQuantityInput}
      handleCheckOut={handleCheckOut}
    />
  </div>
  )
}

export default CartBar;