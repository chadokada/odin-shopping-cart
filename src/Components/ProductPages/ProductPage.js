import React from 'react';
import '../../Styles/App.css';
import Nav from '../Nav';
import ProductCard from '../ProductCard';

const ProductPage = ({title, products, handleAddToCart}) => {
  // PC = proper case

  return (
    <div className='shop-container'>
      {/*
      <TopBar />
      */}
      <div className='shop-content'>
        <Nav shopName={title} />
        <div className='products-container'>

          {products.map((product) => {
            return(
              <ProductCard
                key={product.name} 
                product={product}
                handleAddToCart={handleAddToCart}
              />
            )
          })}

        </div>
      </div>

    </div>
  )
}

export default ProductPage;