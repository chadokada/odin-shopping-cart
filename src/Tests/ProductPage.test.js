import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { products } from '../Components/ProductListing';
import ProductPage from '../Components/ProductPages/ProductPage';

it('displays all products correctly', () => {
  const { container } = render(
    <ProductPage 
      title='Hats'
      products={products.hats}
    />,
    {wrapper: MemoryRouter}
  );

  expect(container).toMatchSnapshot();
})