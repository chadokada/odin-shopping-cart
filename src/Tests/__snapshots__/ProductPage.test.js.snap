// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays all products correctly 1`] = `
<div>
  <div
    class="shop-container"
  >
    <div
      class="shop-content"
    >
      <nav
        class="nav"
      >
        <h2>
          Hats
        </h2>
        <ul>
          <a
            href="/products/all"
          >
            <li>
              All Products
            </li>
          </a>
          <a
            href="/products/hats"
          >
            <li>
              Hats
            </li>
          </a>
          <a
            href="/products/bags"
          >
            <li>
              Bags and Totes
            </li>
          </a>
        </ul>
      </nav>
      <div
        class="products-container"
      >
        <div
          class="product-container"
          name="Ribbed Beanie"
          type="hats"
        >
          <div
            class="product-img-container"
          >
            <img
              alt="Ribbed Beanie"
              class="product-img"
              src="ribbed_beanie.jpg"
            />
          </div>
          <div
            class="product-info"
          >
            <div
              class="product-name"
            >
              Ribbed Beanie
            </div>
            <div
              class="price"
            >
              $20.50
            </div>
          </div>
          <button
            class="add-to-cart-btn"
          >
            Add To Cart
          </button>
        </div>
        <div
          class="product-container"
          name="Bowler"
          type="hats"
        >
          <div
            class="product-img-container"
          >
            <img
              alt="Bowler"
              class="product-img"
              src="bowler.jpg"
            />
          </div>
          <div
            class="product-info"
          >
            <div
              class="product-name"
            >
              Bowler
            </div>
            <div
              class="price"
            >
              $25.00
            </div>
          </div>
          <button
            class="add-to-cart-btn"
          >
            Add To Cart
          </button>
        </div>
        <div
          class="product-container"
          name="Pink Princess Bowler"
          type="hats"
        >
          <div
            class="product-img-container"
          >
            <img
              alt="Pink Princess Bowler"
              class="product-img"
              src="pink_bowler.png"
            />
          </div>
          <div
            class="product-info"
          >
            <div
              class="product-name"
            >
              Pink Princess Bowler
            </div>
            <div
              class="price"
            >
              $35.00
            </div>
          </div>
          <button
            class="add-to-cart-btn"
          >
            Add To Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
